### PR TITLE
feat: [EXT-2360] add test for onvaluechanged

### PR DESCRIPTION
**Pull request by Jwhiles**
# Purpose of PR

## PR Checklist

- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Typescript typings are added/updated/not required

### CODE REVIEW

File: test/cypress/integration/reusable/entry-value-changed-callback-test.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export function entryValueChangedCallbackTest(iframeSelector: any) {
+  it('verifies on valueChanged is called when value changes', () => {
+    cy.getSdk(iframeSelector).then(async (sdk) => {
+      const spy = cy.stub()
+      sdk.entry.fields.body.onValueChanged(spy)
+      expect(spy).to.be.calledOnce
+
+      cy.get('@extension').within(() => {
```
Comment (Jwhiles) [inline test/cypress/integration/reusable/entry-value-changed-callback-test.ts:8]: I'm not sure if it makes sense to actually change the entry this way, really we're just testing the extension at that point. Arguably it's enough to test that `onValueChanged` was called on initial render. WDYT?

Comment (FLoppix) [reply test/cypress/integration/reusable/entry-value-changed-callback-test.ts:8]: You when the extension was rendered?

Comment (Jwhiles) [reply test/cypress/integration/reusable/entry-value-changed-callback-test.ts:8]: Yeah, exactly

Comment (FLoppix) [reply test/cypress/integration/reusable/entry-value-changed-callback-test.ts:8]: Then I agree with you that we should test only on render. We should not alter the state of the extension or interact with it.

Comment (kdamball) [reply test/cypress/integration/reusable/entry-value-changed-callback-test.ts]: For clarification, was the original issue that it wasn't even called on initial render?

**Review by FLoppix — APPROVED** [review]

**Review by FLoppix — APPROVED** [review]

**Pull request merged**